server/instrumentation-backend/src/sh/calaba/instrumentationbackend/utils/ActivityLaunchedWaiter.java

```diff
@@ -43,7 +44,9 @@ public void run() {
         while (true) {
             List<MessageQueue.IdleHandler> idleHandlers = getIdleHandlers(instrumentationMessageQueue);
 
-            for (MessageQueue.IdleHandler idleHandler : idleHandlers) {
+            for (int i = 0; i < idleHandlers.size(); i++) {
+                MessageQueue.IdleHandler idleHandler = idleHandlers.get(i);
```

Comment:
Assuming the final value is removed between lines 47 and 48, `get(i)` will be out of bounds. In Java, this throws an `IndexOutOfBoundsException`. We should catch it and continue if so.